client/src/components/CreateNewBoardModal.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useContext, useRef, useEffect } from 'react';
+import styled from 'styled-components';
+import avatar from '../icons/avatar.jpg';
+import background from '../icons/teamBackground.jpg';
+import { DialogContext, ModalTypes } from '../contexts/DialogContext';
+
+const CreateNewBoardModal = () => {
+    const context = useContext(DialogContext);
+    const modalRef = useRef<HTMLDivElement>(null);
+
+    const onClickOutside = (e: any) => {
+        const element = e.target;
+        if (modalRef.current && !modalRef.current.contains(element)) {
+            e.preventDefault();
+            e.stopPropagation();
+            context.closeModalByType();
+        }
+    };
+
+    useEffect(() => {
+        document.body.addEventListener('click', onClickOutside);
+    });
```

Comment:
If you use `useEffect` without a dependencies array, the effect will be run everytime the component renders. In this case, the event listener will be added multiple times.
Use an empty array (`[]`) to add this listener only once, when component finishes its first render.